### PR TITLE
Pale3 xselpaste

### PR DESCRIPTION
**Pull request by glacambre**
I'm bad at git and accidentally killed #1095 . It looks like I can't push to @pale3's master anymore since the PR is closed and so I had to open this PR instead. Sorry @pale3 :(.

As I said in #1095 , I rebased the PR on top of master. Since lots of changes recently happened regarding the editor/clipboard functions, I had to move things around a bit. The good news is that `clipboard xselpaste` should now probably also work with (simple) contentEditable element.

### CODE REVIEW

File: src/lib/config.ts [hunk 1]
```diff
@@ -95,7 +95,6 @@ class default_config {
      * They consist of key sequences mapped to ex commands.
      */
     ignoremaps = {
-        "<S-Insert>": "mode normal",
```
Comment (bovine3dom) [inline src/lib/config.ts]: Is this supposed to be removed?

Comment (glacambre) [reply src/lib/config.ts]: Ah, probably not, I didn't look at the changes in config.ts (I thought they had been reverted).

File: src/lib/config.ts [hunk 2]
```diff
@@ -110,6 +110,7 @@ class default_config {
      * They consist of key sequences mapped to ex commands.
      */
     inputmaps = {
+        "<S-Insert>": "mode normal",
```
Comment (bovine3dom) [inline src/lib/config.ts]: I'm really confused. Why would shift insert put you in normal mode from insert / input?

I don't think this PR should be making any changes to the config at all.

Comment (glacambre) [reply src/lib/config.ts]: >Why would shift insert put you in normal mode from insert / input?

I don't know. I don't remember if it was already there or if I added it accidentally when trying to re-add the binding for ignoremaps. I nuked every change to config.ts so everything should be alright now.

**Comment by bovine3dom** [comment]
LGTM. Ta @pale3, @glacambre.

**Pull request merged**